app/components/Approvals/TransactionApproval/TransactionApproval.test.tsx

```diff
@@ -107,4 +121,26 @@ describe('TransactionApproval', () => {
 
     expect(wrapper).toMatchSnapshot();
   });
+
+  it('returns null if redesign is enabled', () => {
+    mockApprovalRequest({
+      type: ApprovalTypes.TRANSACTION,
+      // TODO: Replace "any" with type
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } as any);
+
+    (
+      useConfirmationRedesignEnabled as jest.MockedFn<
+        typeof useConfirmationRedesignEnabled
+      >
+    ).mockReturnValue({
+      isRedesignedEnabled: true,
+    });
+
+    const wrapper = shallow(
+      <TransactionApproval transactionType={TransactionModalType.Dapp} />,
+    );
+
+    expect(wrapper).toMatchSnapshot();
```

Comment:
S: do not use Enzyme but the new `renderWithProvider` from our test utils (ideally update all the file):

```suggestion
    const { toJSON } = renderWithProvider(
      <TransactionApproval transactionType={TransactionModalType.Dapp} />,
      {},
    );
    
    expect(toJSON()).toMatchInlineSnapshot('');
```
Also, given that we expect no component render, you can inline the empty string snapshot here to make the test easier to understand at first read, without need to open snapshot file.